src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier;
+    private final Random random;
+
+    public FigureSupplier(ColorSupplier colorSupplier, Random random) {
+        this.colorSupplier = colorSupplier;
+        this.random = random;
+    }
+
+    public Figure getFigureRandom() {
+        int choice = random.nextInt(5);
+
+        switch (choice) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor().toString(),
+                        random.nextInt(10) + 1);
```

Comment:
why some figures use integers and some use doubles? Also, extract a repeating piece of logic to a private method. Better make 10 a constant. You can also create a private utility method for creation of each type of figure, if the resulting code will look too bulky